components/admin/requests/forms/ReasonForReplacementForm.tsx

```diff
@@ -0,0 +1,217 @@
+  FormHelperText,
+  Box,
+  RadioGroup,
+  Radio,
+  Textarea,
+} from '@chakra-ui/react'; // Chakra UI
+import { ReasonForReplacement as ReasonForReplacementEnum } from '@lib/graphql/types'; // Reason For Replacement Enum
+import { formatDate } from '@lib/utils/format'; // Date formatter util
+import { ReasonForReplacement } from '@tools/components/admin/requests/forms/types'; // Reason For Replacement Type
+
+type ReasonForReplacementProps = {
+  readonly reasonForReplacement: ReasonForReplacement;
+  readonly onChange: (updatedData: ReasonForReplacement) => void;
+};
+
+/**
+ * ReasonForReplacementForm Component for allowing users to edit reason for replacement information.
+ *
+ * @param {ReasonForReplacement} ReasonForReplacement Data Structure that holds all reason for replacement information for a client request.
+ * @param onChange Function that uses the updated values from form.
+ */
+export default function ReasonForReplacementForm({
+  reasonForReplacement,
+  onChange,
+}: ReasonForReplacementProps) {
+  return (
+    <>
+      <FormControl as="fieldset" paddingBottom="24px">
+        <FormLabel as="legend" marginBottom="8px">
+          {'Reason'}
+        </FormLabel>
+        <RadioGroup
+          value={reasonForReplacement.reason}
+          onChange={value =>
+            onChange({
+              ...reasonForReplacement,
+              reason: value as ReasonForReplacementEnum,
+            })
+          }
+        >
+          <Stack>
+            <Radio value={ReasonForReplacementEnum.Lost}>{'Lost'}</Radio>
+            <Radio value={ReasonForReplacementEnum.Stolen}>{'Stolen'}</Radio>
+            <Radio value={ReasonForReplacementEnum.Other}>{'Other'}</Radio>
+          </Stack>
+        </RadioGroup>
+      </FormControl>
+
+      {/* Conditionally render this section if Lost is selected as replacement reason */}
+      {reasonForReplacement.reason === ReasonForReplacementEnum.Lost && (
+        <>
+          <FormControl isRequired paddingBottom="24px">
+            <FormLabel>{`Date`}</FormLabel>
+            <Input
+              type="date"
+              value={
+                reasonForReplacement.lostTimestamp
+                  ? formatDate(new Date(reasonForReplacement.lostTimestamp), true)
+                  : ''
+              }
+              onChange={event => {
+                const updatedlostTimestamp = new Date(reasonForReplacement.lostTimestamp);
+                updatedlostTimestamp.setFullYear(parseInt(event.target.value.substring(0, 4)));
+                updatedlostTimestamp.setMonth(parseInt(event.target.value.substring(5, 7)) - 1);
+                updatedlostTimestamp.setDate(parseInt(event.target.value.substring(8, 10)));
+
+                onChange({
+                  ...reasonForReplacement,
+                  lostTimestamp: Date.parse(updatedlostTimestamp.toString())
+                    ? updatedlostTimestamp
+                    : reasonForReplacement.lostTimestamp,
+                });
```

Comment:
Parses the date string inputted in the form field for the year, month, and day, then updates the timestamp with each while keeping the time the same.